src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const title = 'Who\'s the robot?';
+
+  const [entryData, setEntryData] = useState(chatMessages);
+
+  const [totalLikes, setTotalLikes] = useState(0);
+
+  const updateEntryData = updatedEntry => {
+    const entries = entryData.map(entry => {
+      if (entry.id === updatedEntry.id) {
+        setTotalLikes(totalLikes + (updatedEntry.liked ? 1 : -1));
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    });
+    setEntryData(entries);
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>{title}</h1>
+        <p className='likeCount'>{totalLikes} ❤️s</p>
       </header>
       <main>
         {/* Wave 01: Render one ChatEntry component
         Wave 02: Render ChatLog component */}
+        <ChatLog entries={entryData}
+          onUpdateEntry={updateEntryData}
+        ></ChatLog>
       </main>
     </div>
   );
 };
 
```

Comment:
Great job everyone! You both did well on this project! There wasn't much to correct on this one! Let me know if you need to discuss any of my comments in further detail! Congrats! ✨